File: Ecommerce/Core/Entities/Product.cs
﻿using System.ComponentModel.DataAnnotations;

namespace Core.Entities
{
    public class Product : BaseEntity
    {
        [Required]
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        [Required]
        public string PictureUrl { get; set; }
        
        public ProductType ProductType { get; set; }
        public int ProductTypeId { get; set; }

        public ProductBrand ProductBrand { get; set; }
        public int ProductBrandId { get; set; }
    }

}

File: Client/src/model/_Product.ts
interface Product{
  description: string;
  id: number;
  name: string;
  pictureUrl: string;
  price: number;
  productBrand: {
    id: number;
    name: string;
  },
  productBrandId: number;
  productType: {
    id: number;
    name: string;
  }
  productTypeId: number;
}
